docs/rules/terraform_typed_variables.md

```diff
@@ -0,0 +1,37 @@
+# terraform_typed_variables
+
+Disallow `variable` declarations without type.
+
+## Example
+
+```hcl
+variable "no_type" {
+  default = "value"
+}
+
+variable "enabled" {
+  default     = false
+  description = "This is description"
+  type        = bool
+}
+```
+
+```
+$ tflint
+1 issue(s) found:
+
+Notice: `no_type` variable has no type (terraform_typed_variables)
```

Comment:
If you update severity to warning, you must update this as well.

Reply:
Good catch and this has been updated